pysages/utils/compat.py

```diff
@@ -94,7 +94,10 @@ def dispatch_table(dispatch):
     def has_method(fn, T, index):
         types_at_index = set()
         for sig in fn.methods:
-            typ = sig.types[index]
+            if _plum_version_tuple < (2, 3, 0):
+                typ = sig.types[index]
+            else:
+                typ = sig.signature.types[index]
```

Comment:
It's more robust to do the version checking outside the function we make use of. So if you do something like this
```suggestion
            typ = _signature_types(sig)[index]
```
You can add between lines 89 and 91, the following:
```python
    if _plum_version_tuple < (2, 3, 0):

        def _signature_types(sig):
            return sig.types

    else:
        def _signature_types(sig):
            return sig.signature.types

```

Reply:
Changes incorporated into the latest version, let me know if they looks good to you.